### ISSUE TITLE
E10: AA power gen buffers deplete when turned off and not actively powering anything

### ISSUE THREAD

**Issue by Drytchnath** [EnigmaticaModpacks/Enigmatica10#194]
### Modpack Version

1.11.1

### Describe your issue.

In Actually Additions the power generators will keep burning fuel regardless their internal buffer being full or machines needing power. If turned off using redstone they will then slowly discharge their internal buffer - again, nothing is requesting or draining this power, it happens no matter what. I wasted 32 buckets of fuel in a oil generator before I noticed what it was doing.

Bysco in discord asked to have this submitted in github so that they can bring it up to Ellpeck to see if it is intended behavior and possibly update it.

### Crash Report

NA

### Latest Log

NA

### Have you modified the modpack?

No

### Did the issue happen in singleplayer or on a server?

Singleplayer

### Discord Username

tank8912 (Drytchnath)

**Comment by MuteTiefling** [comment]
Thanks for the report!

Out of curiosity, what are you using for power transfer in this setup?

To test, I've placed down an oil generator and provided it with a bucket of oil. This was allowed to burn until the generator was mostly full on power, then disabled with redstone. The generator has retained the same amount of power since being disabled. 

Similarly, another oil generator was given a single bucket and allowed to run until the internal buffer filled. This stopped operating on its own once the internal power buffer was full, no further fuel was burned.

**Comment by MuteTiefling** [comment]
Worth noting that 1.11.1 has AA 1.3.5, while dev is on 1.3.6. So there's a good chance this was simply fixed already.

**Comment by Drytchnath** [comment]
> Worth noting that 1.11.1 has AA 1.3.5, while dev is on 1.3.6. So there's a good chance this was simply fixed already.

I was using advanced laser relays to route power. I even completely removed the relays to test and it still had power buffer decay when nothing was hooked up.

**Comment by SeooY** [comment]
I can't reproduce the issue in 1.11.1 (with Actually Additions 1.3.5)
https://streamable.com/a2ynfl
Besides consuming oil in 50 mB increments, and fully consuming those 50 mB even if the buffer is full, it works as expected.

**Comment by Drytchnath** [comment]
I found its the laser energy relays draining the oil gen dry when idle. Tested with a stand alone oil gen using empowered canola. Linked to a basic energy cube with a relay. Once the cube is full as well as the oil gen buffer the relay will continuously drain power from the oil gen when idle

**Comment by MuteTiefling** [comment]
My test setup was also using AA energy lasers to transmit power to a few machines. When the machines had nothing to do, the lasers were not transmitting power, and so the loss they impose wasn't an issue. The Generator simply stopped as expected.

I'll add an energy cell and see what it does. Default settings are push/pull for the laser and cell, however... so it's very likely that it's self feeding the battery and thus the energy loss causes a constant drain.

Having two power networks should resolve that though. Generators to battery and battery to machines.

**Comment by MuteTiefling** [comment]
Ok, so I can reproduce it with the mek energy cube. Oddly, setting up the split network I described above doesn't fix it.... I've got the laser from the generators into the cube set to push only, and the output one to pull only. The cube has its sides set to match and auto output is off.

**Comment by theboo** [comment]
Does yeeting power loss fix the issue for us, mayhaps? Tho this sounds like an actual bug that should be sent to AA for sure

**Comment by MuteTiefling** [comment]
Yeah, something is goofy with the way it's interacting with the battery. Seems buggy for sure. Will test a bit more and then relay, just to be sure I'm reporting on the right issue.

**Comment by Flanks255** [comment]
Whats the simplest setup you can reproduce this with? because i cant reproduce it in dev with AA and mekanism.

**Comment by MuteTiefling** [comment]
My last test that was doing it was this
- an oil generator
- a basic mekanism energy cell
- connect the two with AA energy lasers

**Comment by Flanks255** [comment]
ok, so i set up a few setups with oil generators and cubes, some full some empty, the full ones run till their buffer fills up then stop, energy never declines 
![image](https://github.com/user-attachments/assets/ba688cdd-d289-4271-b932-cc83c3d2285d),
the non full ones do the same, but take much longer since the cube has to fill up lol

**Comment by Flanks255** [comment]
afaik nothings been touched in the generators since the port, ill check out the video in a bit.

**Comment by MuteTiefling** [comment]
I don't think it's the generator per se.

 It appears to be more to do with the intended power loss of the lasers and a funky interaction with the energy cube.

It's like the lasers are pulling energy from the cube and sending it right back to the cube at a loss.

**Comment by Flanks255** [comment]
ive been doing some tests, it looks like its a specific interaction between AA lasers and Mek cubes, doesnt have anything to do with the generator. i made a test world with AA IE and Mek, and a 3x3  grid of transfer methods and energy storage lol

**Comment by Flanks255** [comment]
got it. Mekanism converts incoming power to MJ, and the lasers were trying to transfer 2 FE to finish filling the cube, getting 1 deducted for laser loss, Mekanism then refusing the 1 FE cause its too small for 1 MJ, but the laser getting confused since it accepted the simulated transfer, but not the actual transfer lol.

https://github.com/Ellpeck/ActuallyAdditions/commit/9fe3fa1c7cdbad77d0e0773a529432f3f04b879e
Fixed in dev.

**Comment by MuteTiefling** [comment]
awesome! Thanks for the help <3

**Comment by HeatherComputer** [comment]
That's such a funny edgecase interaction - but isn't that bad usage of simulation, as both your simulated and real operations should be the same?

**Comment by Flanks255** [comment]
the difference is because of the energy loss, and at that small amount, the actual transfer amount dipped below 1 MJ, and so mek refused it.

**Comment by HeatherComputer** [comment]
Why not account for the loss in your simulation?

**Comment by Flanks255** [comment]
> Why not account for the loss in your simulation?

the simulation is used here to get "how much should i send" kind of...
the code still needs to check if the send fails later, which it does now.
